screen/Running/RunningScreen.js

```diff
@@ -1,58 +1,73 @@
-import ValueWithUnit from '../../common/components/ValueWithUnit';
-import Timer from './components/Timer';
 import Pause from './components/Pause';
-import AudioController from './audioController';
+import AudioController from './controllers/audioController';
+import GameController from './controllers/gameController';
 import GameView from './components/GameView';
+import Header from './components/Header';
+import Socket from '../../network/socket';
+import COLORS from '../../common/constants/COLORS';
 import { getGameResult } from '../../store/gameSlice';
 
 const RunningScreen = ({ route, navigation }) => {
   const { speed, time } = route.params.gameSetting;
   const conversionRate = 0.277778;
-  const [userDistance, setUserDistance] = useState(0);
-  const [zombieDistance, setZombieDistance] = useState(-500);
-  const [hasGameStarted, setHasGameStarted] = useState(false);
-  const [isWinner, setIsWinner] = useState(false);
-  const [hasGameFinished, setHasGameFinished] = useState(false);
-  const [hasOptionClicked, setHasOptionClicked] = useState(false);
+  const dispatch = useDispatch();
 
+  const { id } = useSelector((state) => state.user);
+  const { allPlayersId } = useSelector((state) => state.room);
+  const { role, mode } = useSelector((state) => state.game);
   const canHearingSoundEffect = useSelector(
     (state) => state.ui.canHearingEffect,
   );
   const canHearingBackgroundMusic = useSelector(
     (state) => state.ui.canHearingBGMusic,
   );
 
-  const locationHistory = useRef([]);
+  const [userDistance, setUserDistance] = useState(0);
+  const [opponentDistance, setOpponentDistance] = useState(-500);
+  const [hasGameStarted, setHasGameStarted] = useState(false);
+  const [isWinner, setIsWinner] = useState(false);
+  const [hasGameFinished, setHasGameFinished] = useState(false);
+  const [hasOptionClicked, setHasOptionClicked] = useState(false);
+  const [userCount, setUserCount] = useState(allPlayersId.length);
+
   const survivalTime = useRef(time);
   const countDown = useRef();
   const intervalId = useRef();
   const tracker = useRef();
+  const locationHistory = useRef([]);
+
+  const { current: socket } = useRef(new Socket(BASE_URL));
   const { current: audioController } = useRef(new AudioController());
+  const { current: gameController } = useRef(
+    new GameController(
+      intervalId,
+      countDown,
+      tracker,
+      locationHistory,
+      audioController,
+    ),
+  );
 
   const speedMeterPerSecond = Math.ceil(conversionRate * speed);
-  const distanceGap = Math.ceil(userDistance - zombieDistance);
+  const distanceGap = Math.ceil(userDistance - opponentDistance);
 
   survivalTime.current = time;
 
   const startRunning = async () => {
     setHasGameStarted(true);
 
-    if (canHearingSoundEffect) {
-      audioController.playSoundEffect();
-    }
-
-    if (canHearingBackgroundMusic) {
-      audioController.playBackgroundMusic();
-    }
+    gameController.gameSoundControll(
```

Comment:
오타가 났습니다

```suggestion
    gameController.gameSoundControl(
```

함수이기 때문에 controlGameSound라는 이름도 좋다고 생각합니다.